### ISSUE TITLE
`ess-describe-object-at-point` window should behave like "*Help*" windo

### ISSUE THREAD

**Issue by biocyberman** [emacs-ess/ESS#1005]
I asked and solved the problem partially over here: https://emacs.stackexchange.com/q/58049/8937
However, I would like to report the issue and hope ESS developers consider. My `elisp` skill is still green so I don't understand clearly all the code in `ess-help.el`. Therefore I couldn't hack more.

**Comment by lionel-** [comment]
Maybe we should go in a different direction. I think the way to solve this sort of problems is to provide commands for sending expressions to the REPL. The output shouldn't be in a new window, but in the inferior buffer.

Also these commands should be sensitive to the position of the cursor. For instance with

```r
mt<POINT>cars$cyl
```

we should inspect `mtcars`. And with

```r
mtcars$c<POINT>yl
```

we should inspect `mtcars$cyl`. I've been using such a setup with various commands (print, head, tail, str, internal-inspect, names, etc) with good success.

**Comment by biocyberman** [comment]
Absolutely agree with sensitivity to position of the cursor. 

I thought about sending the output to REPL too. However, the benefit of 'show and clean up' is also nice, to avoid scrolling the REPL too fast. I find it is also useful that the electric buffer allow cycle through several view modes (`str`, `head` `summary`). This may not work so well if we use the REPL. So, with `delete-window-on buf`, things is almost perfect for me. The only improvement I would like to see is a more flexible way to put the buffer `buf` out. Currently it is put on a vertical split by default. 

By the way, I could never get showing via tooltip to work. It is also really nice and ergonomic to show a tooltip buffer when the cursor is that the variable for a certain delay, say 0.5 sec. A keybinding to cycle though view action (e.g. `cycle-describe-view-at-point`) is still needed.   

The same can be applied for function documentations.

**Comment by lionel-** [comment]
> I find it is also useful that the electric buffer allow cycle through several view modes (str, head summary). This may not work so well if we use the REPL

The Emacs command for sending these R commands to the REPL could cycle in the same way. And then you have all the output in the process buffer that can look at simultaneously.

The idea is that the main mode of interaction with R output should be through the REPL for a consistent experience. This would also mean that we don't have to maintain other modes of interaction.

**Comment by biocyberman** [comment]
> The Emacs command for sending these R commands to the REPL could cycle in the same way. And then you have all the output in the process buffer that can look at simultaneously.

Didn't know that it is possible. Does it mean that if new later view mode prints the output below the previous one? If that happens, we should be very careful because some variable contains really long strings, or list.

**Comment by lionel-** [comment]
Printing of long outputs should be controlled via R options. Anyway, I'd first have to implement these methods, perhaps as an external package, so that others can experiment and evaluate the workflow. I'll mark our discussion as off-topic for now.

**Comment by vspinu** [comment]
If I understand correctly, your problem is that when the function exits it doesn't remove the window, right? This is a bug. I didn't notice it because I always have 2-window horizontal split. So my other window is always there. 

For the record, that function creates a temporary window which disappears on any command except the command involving window navigation (like `other-window`). So the difference with *Help* style buffers is that you don't have to press q. Back then it looked like a nifty idea, but maybe it wasn't.

**Comment by biocyberman** [comment]
> If I understand correctly, your problem is that when the function exits it doesn't remove the window, right? This is a bug. 

Correct. And I fixed that with the hack I wrote on [stackexchange](https://emacs.stackexchange.com/a/58051/8937).

> 
> For the record, that function creates a temporary window which disappears on any command except the command involving window navigation (like `other-window`). 

Apparently this is not working, and the fix is mentioned above. 

>So the difference with _Help_ style buffers is that you don't have to press q. Back then it looked like a nifty idea, but maybe it wasn't.

What I like about `*Help*` buffer is that it seems to have a smart way to figure out where to place the window that is less intrusive. For example, when I have a vertical split (one for R source, one for REPL), `*Help*` would be placed as a horizontal place at the bottom. 

The "auto-quit" feature when the key is not one of the keybindings is nice. I like that.

**Comment by vspinu** [comment]
> Apparently this is not working, and the fix is mentioned above.

What command do you use to navigate to the other window?
